dashboard/apps/prototype/management/commands/cache.py

```diff
@@ -25,6 +26,7 @@ def generate(self):
         """
         generate cache
         """
+        cache_tools.METHOD_CACHE_WRITE_ONLY = True
```

Comment:
so if you can access the *args and **kwargs that are passed in the the method call from the wrapper then you wouldn't need to do this.

```python
@method_cache(timeout=24*60*60)
    def people_costs(self, start_date, end_date, contractor_only=False,
                     non_contractor_only=False, ignore_cached=False):
    ..........

# in method_cache class
class method_cache:
.....
    def __call__(self, method):
        """
        return: a wrapped function
        """
        @wraps(method)
        def wrapper(instance, *args, **kwargs):
            key = self.cache_key(method, instance, *args, **kwargs)
            if not kwargs.get('ignore_cached', False) and key in cache:
                logger.debug('cache found for key %s', key)
                return cache.get(key)

            result = method(instance, *args, **kwargs)
            logger.debug('cache generated for key %s', key)
            cache.set(key, result, self.timeout)
            return result
        return wrapper


project.people_costs(start_date, end_date, ignore_cached=True)
```